source/d3d11/draw_call_tracker.cpp

```diff
@@ -12,28 +13,43 @@ namespace reshade::d3d11
 		{
 			const auto destination_entry = _counters_per_used_depthstencil.find(source_entry.first);
 
-			if (destination_entry == _counters_per_used_depthstencil.end())
+			if (reshade::runtime::depth_buffer_retrieval_mode == reshade::runtime::depth_buffer_retrieval_mode::POST_PROCESS)
 			{
-				_counters_per_used_depthstencil.emplace(source_entry.first, source_entry.second);
-			}
-			else
-			{
-				destination_entry->second.vertices += source_entry.second.vertices;
-				destination_entry->second.drawcalls += source_entry.second.drawcalls;
+				if (destination_entry == _counters_per_used_depthstencil.end())
+				{
+					_counters_per_used_depthstencil.emplace(source_entry.first, source_entry.second);
+				}
+				else
+				{
+					destination_entry->second.vertices += source_entry.second.vertices;
+					destination_entry->second.drawcalls += source_entry.second.drawcalls;
+				}
 			}
 		}
 	}
 
-	void draw_call_tracker::reset()
+	void draw_call_tracker::reset(bool ball)
 	{
 		_counters.vertices = 0;
 		_counters.drawcalls = 0;
-		_counters_per_used_depthstencil.clear();
+
+		if (ball == true)
+		{
+			_counters_per_used_depthstencil.clear();
+			_active_depth_texture.reset();
+		}
+	}
+
+	void draw_call_tracker::set_depth_texture(ID3D11Texture2D* depth_texture)
+	{
+		assert(depth_texture != nullptr);
+
+		_active_depth_texture = depth_texture;
 	}
 
 	void draw_call_tracker::track_depthstencil(ID3D11DepthStencilView* depthstencil)
 	{
-		assert(depthstencil != nullptr);
+		// assert(depthstencil != nullptr);
```

Comment:
Why?

Reply:
In some testings, made Reshade crash. Now, should be ok. Will remove the comment.